i18n-helpers/src/lib.rs

```diff
@@ -652,10 +652,10 @@ pub fn reconstruct_markdown<'a>(
     // Markdown without the padding to remove the effect of these
     // structural elements. Similarly, we don't want extra newlines at
     // the start.
-    let simplified_state = state.map(|state| State {
-        newlines_before_start: 0,
-        padding: Vec::new(),
-        ..state
+    let simplified_state = state.map(|mut state| {
```

Comment:
This does indeed look cleaner. I did not think that I could work on the state as a mutable within map even though the option itself is immutable. This part looks good to me.

Reply:
Good, I kept it! :)